assets/js/components/KeyMetrics/ConversionReportingNotificationCTAWidget.js

```diff
@@ -133,25 +147,44 @@ function ConversionReportingNotificationCTAWidget( { Widget, WidgetNull } ) {
 
 	const handleSelectMetricsClick = useCallback( () => {
 		setValue( KEY_METRICS_SELECTION_PANEL_OPENED_KEY, true );
+	}, [ setValue ] );
 
-		if ( shouldShowCalloutForUserPickedMetrics ) {
-			dismissNewConversionReportingEvents();
-		}
+	const isSelectionPanelOpen = useSelect( ( select ) =>
+		select( CORE_UI ).getValue( KEY_METRICS_SELECTION_PANEL_OPENED_KEY )
+	);
+	const isSavingConversionReportingSettings = useSelect( ( select ) =>
+		select( CORE_USER ).isSavingConversionReportingSettings()
+	);
 
-		if ( haveLostConversionEvents ) {
-			dismissLostConversionReportingEvents();
+	useEffect( () => {
+		if ( isSelectionPanelOpen ) {
+			if (
+				( ! isSavingConversionReportingSettings &&
+					( shouldShowCalloutForNewEvents ||
+						shouldShowCalloutForUserPickedMetrics ) ) ||
+				( shouldShowInitialCalloutForTailoredMetrics &&
+					isSavingConversionReportingSettings )
+			) {
+				dismissCallout();
+			}
```

Comment:
This is quite nested and tough-to-read 😅 

Can you add some comments explaining what each conditional is so it's clearer what's happening here?

Reply:
Good point, I added comments

Reply:
Those are awesome, thanks! 🙏🏻